json_encoder.go

```diff
@@ -38,14 +38,20 @@ const (
 	_initialBufSize = 1024
 )
 
-var jsonPool = sync.Pool{
-	New: func() interface{} {
-		return &jsonEncoder{
-			// Pre-allocate a reasonably-sized buffer for each encoder.
-			bytes: make([]byte, 0, _initialBufSize),
-		}
-	},
-}
+var (
+	// ErrNilEntry signals that Encoder.WriteEntry was called with a nil *Entry.
+	ErrNilEntry = errors.New("can't encode a nil *Entry")
```

Comment:
nit: prefer not exporting these errors. I don't think there's a reasonable way to use these errors in code since it would be a bad code bug.